src/utils/static-if.hpp

```diff
@@ -5,7 +5,7 @@
 #ifndef __pinocchio_utils_static_if_hpp__
 #define __pinocchio_utils_static_if_hpp__
 
-#include "pinocchio/fwd.hpp"
+#include "pinocchio/traits.hpp"
```

Comment:
Use fwd here.

Reply:
`pinocchio/fwd` loads the `Eigen` module as well. I had to separate `traits`, since I need to load `minmax.hpp` before I load `Eigen` module. So `static-if` needs to be independent from `Eigen`, since it is being loaded with `minmax.hpp`.

Reply:
Why do you need to load minmax.hpp before Eigen?

Reply:
For me, there is no apparent reason.

Reply:
The problem is in Line 818 at https://eigen.tuxfamily.org/dox/MathFunctions_8h_source.html.
If Eigen is loaded before, the default impementation of `std::min` is used.

Reply:
Then the inclusion order should not change, it is just the function signature that should be specialized.

Reply:
In addition, this won't be the std version which will be called here, but the one located in CppAD namespace.

Reply:
The clean implementation is to define `min` and `max` in the namespace `CppAD`.

Reply:
I didn't find implementation of `min/max` binary operation in CppAD. I could only find implementation of `numeric_limits::min()/max()`

Reply:
So, you have to implement them in Pinocchio ;)

Reply:
I would suggest to only keep the unit test and the minmax.hpp files and to restart from scratch.

Reply:
I tried to put that earlier, but `std::min` is seen first by Eigen. The template resolves to the default implementation

Reply:
It should not. If min/max is defined in CppAD and included, this should work fine. std::min for CppAD::AD is not defined